internal/jimmhttp/httpproxy_handler.go

```diff
@@ -55,23 +55,31 @@ func (hph *HTTPProxyHandler) ProxyHTTP(w http.ResponseWriter, req *http.Request)
 
 	modelUUID := chi.URLParam(req, "uuid")
 	if modelUUID == "" {
-		writeError(ctx, w, http.StatusUnprocessableEntity, errgo.New("cannot parse path"), "cannot parse path")
+		msg := "cannot parse model UUID from path"
+		writeError(ctx, w, http.StatusBadRequest, errors.E(msg), msg)
 		return
 	}
-	model, err := hph.jimm.JujuManager().GetModel(ctx, modelUUID)
-	if err != nil {
-		writeError(ctx, w, http.StatusNotFound, err, "cannot get model")
+
+	if !names.IsValidModel(modelUUID) {
+		msg := "invalid model UUID format"
+		writeError(ctx, w, http.StatusBadRequest, errors.E(msg), msg)
 		return
 	}
-	u, p, err := hph.jimm.CredentialStore.GetControllerCredentials(ctx, model.Controller.Name)
+
+	controllerDetails, err := hph.jimm.JujuManager().ControllerDetailsForModel(ctx, modelUUID)
 	if err != nil {
-		writeError(ctx, w, http.StatusNotFound, err, "cannot retrieve credentials")
+		if errors.ErrorCode(err) == errors.CodeNotFound {
+			writeError(ctx, w, http.StatusNotFound, err, "controller details not found")
+			return
+		}
+		writeError(ctx, w, http.StatusInternalServerError, err, "failed to get controller details")
```

Comment:
are we giving away implementation details? or should return a vague error about failing to proxy the request and log the error?